esmvaltool_sample_data/sample_data.py

```diff
@@ -112,6 +109,26 @@ def select_host(hosts, preferred_hosts, ignore_hosts):
     return hosts[0]
 
 
+def select_latest_versions(datasets: dict) -> dict:
+    """Return a dict with only the latest version of each dataset.
+
+    If `datasets` contains multiple versions of a dataset, return only
+    the most recent version.
+    """
+    keys = (key.rsplit('.', 1) for key in list(datasets.keys()))
+    keys = sorted(keys)
+    grouped = groupby(keys, key=lambda key: key[0])
+
+    most_recent_keys = (list(grouper)[-1] for group, grouper in grouped)
```

Comment:
```suggestion
    most_recent_keys = (list(versions)[-1] for group, versions in grouped)
```
might be a more readable name?